src/hotspot/share/jfr/jni/jfrJavaSupport.cpp

```diff
@@ -759,72 +759,58 @@ bool JfrJavaSupport::is_excluded(jobject thread) {
   return native_thread != NULL ? native_thread->jfr_thread_local()->is_excluded() : is_thread_excluded(thread);
 }
 
+static const Klass* get_handler_field_descriptor(const Handle& h_mirror, fieldDescriptor* descriptor, TRAPS) {
+  assert(h_mirror.not_null(), "invariant");
+  assert(descriptor != NULL, "invariant");
+  Klass* const k = java_lang_Class::as_Klass(h_mirror());
+  assert(k->is_instance_klass(), "invariant");
+  InstanceKlass* const ik = InstanceKlass::cast(k);
+  if (ik->is_not_initialized()) {
+    ik->initialize(CHECK_NULL);
+  }
+  assert(ik->is_being_initialized() || ik->is_initialized(), "invariant");
+  const Klass* const typed_field_holder = ik->find_field(vmSymbols::eventHandler_name(),
+                                                         vmSymbols::jdk_jfr_internal_handlers_EventHandler_signature(),
+                                                         true,
+                                                         descriptor);
+  return typed_field_holder != NULL ? typed_field_holder : ik->find_field(vmSymbols::eventHandler_name(),
+                                                                          vmSymbols::object_signature(), // untyped
```

Comment:
When would be the event handler field untyped?

Reply:
For events in java.base.